### PR TITLE
Feature/blog front end

### PR DESCRIPTION
**Pull request by patelsaajan**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,6 +29,7 @@
     "firebase-admin": "^11.11.1",
     "framer-motion": "^11.0.14",
     "lodash": "^4.17.21",
+    "markdown-to-jsx": "^7.4.6",
```
Comment (finnformica) [inline package.json:32]: are you still using markdown-to-jsx or react-markdown? if not could you uninstall deps pls

File: src/app/(landing-page)/blog/[posts]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        setLoadedDoc(docSnap.data() as IBlog);
+      } else {
+        notFound();
+      }
+    };
+
+    getBlog();
+  }, [params.posts]);
+
+  if (loadedDoc) {
```
Comment (finnformica) [inline src/app/(landing-page)/blog/[posts]/page.tsx:33]: could you change this to if (!loadedDoc) return Loading, just to reduce nested code

File: src/app/(landing-page)/blog/page.tsx [hunk 3]
```diff
@@ -14,20 +14,26 @@ const BlogPage = () => {
   if (posts !== undefined) {
```
Comment (finnformica) [inline src/app/(landing-page)/blog/page.tsx:14]: likewise here, if (!posts) return Loading

File: src/components/BlogForm/AddBlog.tsx [hunk 4]
```diff
@@ -58,7 +58,7 @@ const AddBlog = () => {
   }
   const { user } = session;
 
-  const onSubmit = (data: BlogFormType) => {
+  const onSubmit = (data: IBlogForm) => {
     if (!heroPhoto) {
       return enqueueSnackbar("No here image selected.", { variant: "error" });
```
Comment (finnformica) [inline src/components/BlogForm/AddBlog.tsx:63]: typo

File: src/components/BlogPage/BlogCard.tsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+            ) : (
+              <Skeleton variant="rectangular" width={345} height={140} />
+            )}
+          </Stack>
+        </CardActionArea>
+        <CardActions>
+          <Stack direction={"row"} gap={2} sx={{ alignContent: "center" }}>
+            <Chip label={author} variant="outlined" color="primary"></Chip>
+            <Chip label={tags} variant="outlined" color="primary" />
+            <Chip label={date} variant="outlined" color="primary"></Chip>
```
Comment (finnformica) [inline src/components/BlogPage/BlogCard.tsx:91]: empty tags should be self-closing

File: src/components/BlogPage/BlogPost.tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
+import { Box, Skeleton, Stack, Typography } from "@mui/material";
+import { getDownloadURL, ref } from "firebase/storage";
+import { storage } from "lib/firebase/config";
+import Image from "next/image";
+import { useEffect, useState } from "react";
+import { IBlog } from "./types";
+
+const chosenFontSize = 14;
```
Comment (finnformica) [inline src/components/BlogPage/BlogPost.tsx:8]: global constants in files should be all caps, DEFAULT_FONT_SIZE = 14

File: src/components/BlogPage/types.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+export interface IBlog {
+  author: string;
+  content?: string;
```
Comment (finnformica) [inline src/components/BlogPage/types.ts:3]: should content and summary be optional?

**Review by finnformica — APPROVED** [review]
generally looks good !

File: src/components/BlogPage/BlogBack.tsx [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import { Button } from "@mui/material";
+import ArrowBackIcon from "@mui/icons-material/ArrowBack";
+
+const BlogBack = () => {
+  return (
+    <Button sx={{ my: 2 }} href="/blog">
+      <ArrowBackIcon />
```
Comment (finnformica) [inline src/components/BlogPage/BlogBack.tsx:7]: Use the button prop 'startIcon'

`<Button startIcon={<ArrowBack />}>
Back
</Button>`

also does using the href like that cause the page to reload when navigating? if so, could you use the router from next/navigation and `router.push('/blog')`

File: src/components/BlogPage/BlogCard.tsx [hunk 9]
```diff
@@ -46,52 +47,58 @@ const BlogCard = ({
...
-            <Chip label={author} variant="outlined" color="primary"></Chip>
-            <Chip label={tags} variant="outlined" color="primary" />
-            <Chip label={date} variant="outlined" color="primary"></Chip>
-          </Stack>
-        </CardActions>
       </Link>
+      <CardActions sx={{ p: 2 }}>
+        <Stack direction={"row"} gap={2} sx={{ alignContent: "center" }}>
+          <Chip label={tags} variant="outlined" color="primary" />
+          <Chip label={date} variant="outlined" color="primary"></Chip>
```
Comment (finnformica) [inline src/components/BlogPage/BlogCard.tsx:99]: self-closing

File: src/components/BlogPage/BlogCard.tsx [hunk 10]
```diff
@@ -46,52 +47,58 @@ const BlogCard = ({
...
-            title={title}
-            sx={{
-              height: 50,
-              overflow: "hidden",
-            }}
+            title={
+              <Stack direction={"row"} gap={1}>
+                <Typography variant="body2">{author}</Typography>
+                <Typography variant="body2">·</Typography>
+                <Typography variant="body2" color={"grey"}>
```
Comment (finnformica) [inline src/components/BlogPage/BlogCard.tsx:58]: use color="text.secondary" to be consistent with the rest of the application

File: src/components/BlogForm/AddBlog.tsx (hunk 4 above)
Comment (patelsaajan) [reply src/components/BlogForm/AddBlog.tsx:63]: typo? I for interface

File: src/components/BlogPage/types.ts (hunk 7 above)
Comment (patelsaajan) [reply src/components/BlogPage/types.ts:3]: so was meant to get your ops, basically i re use that type one for the card and one for the page. but the difference is one uses the summary and one uses the content

Comment (finnformica) [reply src/components/BlogPage/types.ts:3]: hmmm, i would have a card type and then the blog type extends that and adds the extra content

surely both use summary if its stored in the same object on firestore?

**Pull request merged**